### PR TITLE
L1 Profile update issue fixes

### LINKED ISSUE #141: dsVideoDevice : L1 issues for profile reading
* Frame rate in profile is defined as string, and there are some parsing and logical issues identified associated with this.
* Checking of support for API in source or sink device for done before calling the API which causes result check with NOT_SUPPORTED failing in L1
* There are few cases in which NULL is type casted for comparison purposes, this was working in the case of source device but failing for Sink device.

### PR DESCRIPTION
**Pull request by akhilbhas**
* updatign function call logic to address the result check properly

closes #141

### CODE REVIEW

**Review by thanushreevani — APPROVED** [review]

File: src/test_dsVideoDevice_parse_configuration.h [hunk 1]
```diff
@@ -83,7 +83,7 @@ typedef struct _dsVideoDeviceConfiguration_t {
     int32_t  HDRCapabilities;
     uint32_t SupportedVideoCodingFormats;
     int32_t  NoOfSupportedDFR;
-    char     SupportedDisplayFramerate[dsVIDEO_FRAMERATE_MAX];
+    char     SupportedDisplayFramerate[dsVIDEO_FRAMERATE_MAX][DS_VIDEO_DEVICE_KVP_SIZE];
```
Comment (hari22yuva) [inline src/test_dsVideoDevice_parse_configuration.h:86]: Update this 
diff --git a/src/test_dsVideoDevice_parse_configuration.h b/src/test_dsVideoDevice_parse_configuration.h
index 3454e1f..b03f8b5 100644
--- a/src/test_dsVideoDevice_parse_configuration.h
+++ b/src/test_dsVideoDevice_parse_configuration.h
@@ -69,8 +69,8 @@

 #define DS_VIDEO_DEVICE_KVP_SIZE              128
 #define DS_VIDEO_DEVICE_PORT_NAME_SIZE        32
-#define DS_VIDEO_DEVICE_MAX_MS12_PROFILE_LIST 10
-#define DS_VIDEO_DEVICE_MAX_MS12_PROFILE_LEN  32
+#define DS_VIDEO_DEVICE_MAX_FRAMERATE_LIST 10
+#define DS_VIDEO_DEVICE_MAX_FRAMERATE_LEN  32
 #define DS_VIDEO_DEVICE_NAME_SIZE             64

 #define DS_VIDEO_DEVICE_MODULE_NAME    "dsVideoDevice"
@@ -83,7 +83,7 @@ typedef struct _dsVideoDeviceConfiguration_t {
     int32_t  HDRCapabilities;
     uint32_t SupportedVideoCodingFormats;
     int32_t  NoOfSupportedDFR;
-    char     SupportedDisplayFramerate[dsVIDEO_FRAMERATE_MAX];
+    char     SupportedDisplayFramerate[DS_VIDEO_DEVICE_MAX_FRAMERATE_LIST][DS_VIDEO_DEVICE_MAX_FRAMERATE_LEN];
     int32_t  num_codec_entries;
     float    level;
     dsVideoCodecHevcProfiles_t  profile;

**Review by thanushreevani — APPROVED** [review]

File: src/test_l1_dsVideoDevice.c [hunk 2]
```diff
@@ -394,15 +393,18 @@ void test_l1_dsVideoDevice_positive_dsSetDFC (void)
     for (int i = 0; i < gDSvideoDevice_NumVideoDevices; ++i) {
         result = dsGetVideoDevice(i, &handle);
         UT_ASSERT_EQUAL(result, dsERR_NONE);
-        UT_ASSERT_NOT_EQUAL(handle, (intptr_t)NULL);
-        if(gSourceType == 1) {
-            // 03: Set DFC mode with various zoom modes
-            for (int j = 0; j < gDSVideoDeviceConfiguration[j].NoOfSupportedDFCs; ++j) {
-                result = dsSetDFC(handle, gDSVideoDeviceConfiguration[i].SupportedDFCs[j]);
+        // 03: Set DFC mode with various zoom modes
+        for (int j = 0; j < gDSVideoDeviceConfiguration[j].NoOfSupportedDFCs; ++j)
```
Comment (hari22yuva) [inline src/test_l1_dsVideoDevice.c:397]: Update `i`
for (int j = 0; j < gDSVideoDeviceConfiguration[i].NoOfSupportedDFCs; ++j)

File: src/test_l1_dsVideoDevice.c [hunk 3]
```diff
@@ -513,13 +514,11 @@ void test_l1_dsVideoDevice_positive_dsGetDFC(void)
...
-        UT_ASSERT_NOT_EQUAL(handle, (intptr_t)NULL);
 
         // Step 03: Get the DFC mode using dsGetDFC() with the obtained handle
+        result = dsGetDFC(handle, &dfc_mode);
         if(gSourceType == 1) {
-            result = dsGetDFC(handle, &dfc_mode);
             UT_ASSERT_EQUAL(result, dsERR_NONE);
-
             // Step 04: Compare the DFC mode with value from the profile
             for (int j = 0; j < gDSVideoDeviceConfiguration[j].NoOfSupportedDFCs; ++j) {
```
Comment (hari22yuva) [inline src/test_l1_dsVideoDevice.c:523]: update `i`
for (int j = 0; j < gDSVideoDeviceConfiguration[i].NoOfSupportedDFCs; ++j) {

check in L2 code also 
line:122

**Review by hari22yuva — COMMENTED** [review]
Please address the review comments

File: src/test_l1_dsVideoDevice.c (hunk 3 above)
Comment (akhilbhas) [reply src/test_l1_dsVideoDevice.c]: Done

File: src/test_l1_dsVideoDevice.c (hunk 2 above)
Comment (akhilbhas) [reply src/test_l1_dsVideoDevice.c]: Done

File: src/test_l2_dsVideoDevice.c [hunk 4]
```diff
@@ -392,7 +392,7 @@ void test_l2_dsVideoDevice_SetAndVerifyDisplayframerate_sink(void)
             ret = dsGetCurrentDisplayframerate(handle, getframerate);
             UT_ASSERT_EQUAL(ret, dsERR_NONE);
             UT_LOG_DEBUG("dsGetCurrentDisplayframerate return status: %d, framerate: %s", ret, getframerate);
-            UT_ASSERT_STRING_EQUAL(gDSVideoDeviceConfiguration[i].SupportedDisplayFramerate, getframerate);
+            UT_ASSERT_STRING_EQUAL(gDSVideoDeviceConfiguration[i].SupportedDisplayFramerate[j], getframerate);
         } /* for(j) */
     } /* for(i) */
```
Comment (hari22yuva) [inline src/test_l2_dsVideoDevice.c:398]: Add one more test case in L2 with 
dsSetFRFMode() & dsGetFRFMode()

Comment (akhilbhas) [reply src/test_l2_dsVideoDevice.c:398]: Added

File: profiles/sink/Sink_VideoDevice.yaml [hunk 5]
```diff
@@ -36,7 +36,5 @@ dsVideoDevice:
       #dsVIDEO_CODEC_MAX         = (0x01 << 3)    /*!< Out of range */
       SupportedVideoCodingFormats: 0x06  #dsVIDEO_CODEC_MPEG2|dsVIDEO_CODEC_MPEG4PART10
       SupportedDisplayFramerate:
-        - "60"
-        - "50"
-        - "24"
+        - "3840x2160px60"
```
Comment (hari22yuva) [inline profiles/sink/Sink_VideoDevice.yaml:39]: Please check this description and update the supported frame rates 
https://github.com/rdkcentral/rdk-halif-device_settings/blob/develop/include/dsVideoDevice.h#L409

File: src/test_l2_dsVideoDevice.c (hunk 4 above)
Comment (hari22yuva) [reply src/test_l2_dsVideoDevice.c:398]: Can you update the 
ds-video-device_L2-Low-Level_TestSpec.md

**Review by hari22yuva — COMMENTED** [review]
please check the comments

**Review by thanushreevani — DISMISSED** [review]

File: profiles/sink/Sink_VideoDevice.yaml (hunk 5 above)
Comment (krishna-1809) [reply profiles/sink/Sink_VideoDevice.yaml:46]: Supported FrameRates for 4K platforms:

3840x2160px47
3840x2160px48
3840x2160px50
3840x2160px59
3840x2160px60

Supported FrameRates for 2K:
1366X768px47
1366X768px48
1366X768px50
1366X768px59
1366X768px60

File: src/test_l2_dsVideoDevice.c (hunk 4 above)
Comment (akhilbhas) [reply src/test_l2_dsVideoDevice.c:398]: updated

File: profiles/sink/Sink_VideoDevice.yaml (hunk 5 above)
Comment (akhilbhas) [reply profiles/sink/Sink_VideoDevice.yaml]: updated

**Review by hari22yuva — APPROVED** [review]
looks ok

**Review by thanushreevani — APPROVED** [review]

**Pull request merged**